packages/workspace/src/browser/workspace-trust-service.ts

```diff
@@ -115,17 +115,19 @@ export class WorkspaceTrustService {
     }
 
     protected async handlePreferenceChange(change: PreferenceChange): Promise<void> {
-        if (change.preferenceName === WORKSPACE_TRUST_STARTUP_PROMPT && change.newValue !== WorkspaceTrustPrompt.ONCE) {
-            this.storage.setData(STORAGE_TRUSTED, undefined);
-        }
+        if (change.scope === PreferenceScope.User) {
```

Comment:
Suggestion: This change isn't necessary. The change event will not trigger if a `User` scoped preference is changed in the `Workspace` scope.

Reply:
It's the second line of defense :)

Reply:
Ok, fine with me.